### ISSUE TITLE
QoL changes from Hellfire v1.02.A5

### ISSUE THREAD

**Issue by Deathrizes** [diasurgical/devilutionX#1136]
Many great changes and additions for hellfire that makes the game much more enjoyable!

-[   ] Item Outlines: In the dungeon, the color drawn to outline items on the floor now match the type of item (e.g. Gold outline = unique item).
-[   ] Adria and Pepin automatically fill your mana and life, no longer needing an extra action.
-[   ] Spell Scrolls: Unlearnable spells (Identify, Infravision, Resurrect) have blue ribbons instead of red ones.
-[   ] Durability Warnings: Now appear 2 points earlier (yellow and red at 7 and 4, instead of 5 and 2).
-[   ] Gold Piles: Always up to 10k (Personally 50k which I seen many suggest seems better, less clutter).
-[   ] Town Respawn: You now respawn within sight of Town Portals (rejoin other players faster).
-[   ] More Player Information: To improve coop play, mousing over other players now additionally shows their Armor Class and Resistances.
-[   ] Maxed Resistances: Now display their value (75%) instead of MAX for better display consistency with the above feature.
-[   ] Griswold: Stops favoring the sale of item types you don't have (now I can stop calling him "Grisbald the Junksmith").
-[   ] Wirt: Stops favoring the sale of item types you don't have (i.e. if you carry 3 bows, he'd virtually never sell a bow).
-[   ] Blocked Doors: No longer highlighted for interaction by the mouse; it can no longer obscure items behind it.
-[   ] New Game/Quit: To prevent accidental exits, these options are now disabled while you are dead in a multiplayer game.
-[   ] Chain Lightning: No longer scans for targets further than the corners of your screen (reduces gapping).
-[   ] Defiler Speeches: You will no longer hear the Defiler speak before you actually encounter it (no more awful disembodied taunting).

**Comment by AJenbo** [comment]
I corrected the spacing so that the list now appears correct with usable a checkmark :)

Pepin automatically healing you is actually a normal feature of Hellfire 1.00, so this is already a part of DevilutionX.
Adria never refilled mana for free so that is a bit different.

**Comment by malvarenga123** [comment]
>Gold Piles: Always up to 10k (Personally 50k which I seen many suggest seems better, less clutter).
I would like a Diablo 2 approach where gold is just a number in the inventory

**Comment by NeonLightning** [comment]
i gotta say all these changes would be nice(although i don't play multiplayer)

**Comment by julealgon** [comment]
The change to durability warnings make zero sense to me. I don't see the reason behind it at all.

**Comment by galaxyhaxz** [comment]
I agree with the scroll variants, there are 3 and we can guess they intended to have them for lightning, magic, and fire.
![screen00](https://user-images.githubusercontent.com/65138215/110942271-34db7100-82ff-11eb-949a-45304930e169.png)

**Comment by julealgon** [comment]
> ... there are 3 ...

Wait, do you mean there are unused graphics for them @galaxyhaxz ?

That screenshot looks really cool.

**Comment by qndel** [comment]
applying trns to items probably wouldn't be very hard and it'd allow you to have as many colors as you want without new graphics ;p

**Comment by julealgon** [comment]
> applying trns to items probably wouldn't be very hard and it'd allow you to have as many colors as you want without new graphics ;p

Yeah but are those scrolls TRN swaps that exist somewhere, custom graphics, or just completely done manually?

If they are already there but unused, it is much easier to repurpose.

**Comment by galaxyhaxz** [comment]
TRN for items will be cool, and it's easy to generate them with code to prevent extra files.

The other scrolls are unused graphics already in the .CEL file. There are quite a few, actually.

**Comment by GrayFace** [comment]
> * [ ]  New Game/Quit: To prevent accidental exits, these options are now disabled while you are dead in a multiplayer game.

Misclicking New Game instead of Load is also frequent in single play for me. I imagine spacing menu buttons out and making space between buttons not count as one of them should be a good remedy.

**Comment by AJenbo** [comment]
maybe a confirm option?

**Comment by GrayFace** [comment]
A confirmation for New Game and Quit, but no confirmation for Load Game? Sounds good, I like it.
Also, thinking of it, the ideal thing for single player would be quick Save/Load keys.

**Comment by julealgon** [comment]
>  the ideal thing for single player would be quick Save/Load keys

That would be a good addition indeed, however, in all games that had save and load keys next to each other, I always ended up saving instead of loading and vice versa by accident 🤣 

That's probably on me though... however, I wonder if the shortcuts could be a bit more "semantic" to avoid this problem (maybe make "save" more accessible than "load" or something like that).

**Comment by GrayFace** [comment]
> > the ideal thing for single player would be quick Save/Load keys
> 
> That would be a good addition indeed, however, in all games that had save and load keys next to each other, I always ended up saving instead of loading and vice versa by accident

Yes, I remember this issue in Fallout, which didn't allow configuring them. I always set them to F5 and F8, I think Evil Islands had them like that and I'm using it in every game since then.

**Comment by qndel** [comment]
> Griswold: Stops favoring the sale of item types you don't have (now I can stop calling him "Grisbald the Junksmith").
> Wirt: Stops favoring the sale of item types you don't have (i.e. if you carry 3 bows, he'd virtually never sell a bow).
I actually consider these changes to be detrimental for the player as they remove a way to manipulate the type of shop items,. All it takes is unequpping (dropping on the ground?) your items if you want them to roll all items xD
